### PR TITLE
PR for SorrTask395_Unit_test_plotting_gallery_notebook

### PR DESCRIPTION
**Pull request by dchoi127**
Completed unit test using dummy config builder from research_amp.cc.algotrading. Ran the unit test and it completed successfully. Wasn't sure what files to add for the outcome of the unit test. 

For some strange reason when I rerun the tests they seem to hang.

Proof that it ran successfully before:

Screenshot 2023-07-07 at 3 13 21 PM

Now:
Screenshot 2023-07-07 at 3 13 47 PM

It just hangs and when I terminate the command it says "no tests ran in...", I don't believe I changed anything

### CODE REVIEW

**Comment by samarth9008** [comment]
Name your PR according to your task.

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        """
+        # Run the notebook.
+        hsystem.system(cmd)
+
+
+def build_config_list() -> cconfig.ConfigList:
+    """
+    Build the config to run the notebook.
+    """
+    config = ramccalg.get_default_config()
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:37]: actually this command creates a config that is not related to this task and it is also being saved above
we do not need it

instead use the approach like at `amp/im_v2/common/notebooks/test/test_master_raw_data_gallery_notebook.py`
create same config and inherit your test class from `dsnrn.Test_Run_Notebook_TestCase`

File: core/plotting/notebooks/test/outcomes/result_0/config.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+client_config: 
+  resample_1min: False
+  tag: downloaded_1sec
+  contract_type: futures
+  universe:
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/outcomes/result_0/config.txt:5]: remove this file

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import core.config as cconfig
+import helpers.hsystem as hsystem
+import helpers.hunit_test as hunitest
+import research_amp.cc.algotrading as ramccalg
+
+_LOG = logging.getLogger(__name__)
+
+
+class TestSorrTask393Notebook(hunitest.TestCase):
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:13]: lets call it just `TestGalleryNotebook`

**Comment by dchoi127** [comment]
Pytest seems to work again. Proof that test runs without errors 

Screenshot 2023-07-10 at 11 37 49 AM

**Comment by DanilYachmenev** [comment]
See the follow-up comments and implement here https://github.com/sorrentum/sorrentum/pull/393#pullrequestreview-1522601550

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        """
+        notebook_path = "core/plotting/notebooks/gallery_notebook.ipynb"
+        config_func_path = "core.plotting.notebooks.test.test_gallery_notebook.build_config_list()"
+        # Run the notebook.
+        self._test_run_notebook(notebook_path, config_func_path)
+
+
+def build_config_list() -> cconfig.ConfigList:
+    """
+    Simple config builder for the test.
```
Comment (gpsaggese) [inline core/plotting/notebooks/test/test_gallery_notebook.py:24]: Empty config builder used by the test.

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    # We want to execute the notebook as it is, but config builder needs
```
Comment (gpsaggese) [inline core/plotting/notebooks/test/test_gallery_notebook.py:26]:
```
# We want to execute the notebook as it is, but a config is needed so we pass an empty one.
```

File: core/plotting/notebooks/gallery_notebook.py [hunk 6]
```diff
@@ -57,12 +54,6 @@
...
-# Set inputs.
-seq = np.concatenate([np.random.uniform(-1, 1, 100), np.random.choice([5, 10], 100)])
-index = pd.date_range(start="2023-01-01", periods=len(seq), freq="D")
-srs = pd.Series(seq, index=index)
-lag = 7
-# TODO(Dan): Remove after integration with `cmamp`
-figsize = (20,20)
-# Plot.
-cpvistte.plot_histograms_and_lagged_scatterplot(srs, lag, figsize=figsize)
+test_plots.Test_plots.test_plot_histograms_and_lagged_scatterplot1()
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/gallery_notebook.py:57]: no, you cant do it this way
methods that implement the actual testing are only for unit tests

here https://github.com/sorrentum/sorrentum/pull/393#discussion_r1258625891 you're asked to create a helper function that only returns a series that will be an input

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    # We want to execute the notebook as it is, but config is needed
+    # so we pass an empty one.
+    config = {}
+    config = cconfig.Config()
+    config_list = cconfig.ConfigList([config])
+    return config_list
+
+@pytest.mark.superslow("~40 sec.")
+class TestGalleryNotebook(dsnrn.Test_Run_Notebook_TestCase):
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:22]: Comment for integrators, do not resolve

Not sure if it is going to work now due to [#4765](https://github.com/cryptokaizen/cmamp/issues/4765)
Probably will need to pause until the mentioned issue is fixed

@dchoi127 still implement the code so it is ready to merge when the bug is fixed
when comments are implemented we can check this guess for sure

File: core/plotting/test/test_plots.py [hunk 8]
```diff
@@ -10,18 +10,18 @@
...
         # Set inputs.
-        seq = np.concatenate([np.random.uniform(-1, 1, 100), np.random.choice([5, 10], 100)])
-        index = pd.date_range(start="2023-01-01", periods=len(seq), freq="D")
-        srs = pd.Series(seq, index=index)
+        rng = np.random.default_rng(seed=0)
+        samples = rng.normal(size=100)
+        index = pd.date_range(start="2023-01-01", periods=len(samples), freq="D")
+        srs = pd.Series(samples, index=index)
         lag = 7
         # TODO(Dan): Remove after integration with `cmamp`
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:24]: Following https://github.com/sorrentum/sorrentum/pull/393#discussion_r1259833216
-> 
```
# TODO(Dan): Remove after integration with `cmamp`. Changes from Cm #4722 are not in `sorrentum` yet.
```

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        """
+        notebook_path = "core/plotting/notebooks/gallery_notebook.ipynb"
+        config_func_path = "core.plotting.notebooks.test.test_gallery_notebook.build_config_list()"
+        # Run the notebook.
+        self._test_run_notebook(notebook_path, config_func_path)
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (DanilYachmenev) [reply core/plotting/notebooks/test/test_gallery_notebook.py:35]: linter should fix it

File: core/plotting/notebooks/gallery_notebook.py (hunk 6 above)
Comment (dchoi127) [reply core/plotting/notebooks/gallery_notebook.py:57]: Just as clarfication.

So the helper function should be created inside `core/plotting/test/test_plots.py` (as a separate static function). Then I should call the helper function inside the notebook (to get inputs) and then plot them?

Comment (dchoi127) [reply core/plotting/notebooks/gallery_notebook.py:57]: Additionally, if I am not mistaking the helper function should return all the inputs: series, lags, figsize (as a tuple). Rather than just a series?

Comment (DanilYachmenev) [reply core/plotting/notebooks/gallery_notebook.py:57]: as to the function - yes
however, lets do it now even as a separate function
until this comment is answered https://github.com/sorrentum/sorrentum/pull/393#discussion_r1260051375

in general I'd really like to avoid initiating a test class and call its method in the notebook

Comment (DanilYachmenev) [reply core/plotting/notebooks/gallery_notebook.py:57]: as to return - just series, since this is what requires computations
lag, figsize etc are just additional params for the call

Comment (dchoi127) [reply core/plotting/notebooks/gallery_notebook.py:57]: > as to the function - yes however, lets do it now even as a separate function until this comment is answered [#393 (comment)](https://github.com/sorrentum/sorrentum/pull/393#discussion_r1260051375)
> 
> in general I'd really like to avoid initiating a test class and call its method in the notebook

Sorry just to be clear one more time. Until the comment is answered the notebook should not use the helper method and just manually set the inputs like original?

File: core/plotting/notebooks/gallery_notebook.py [hunk 10]
```diff
@@ -54,6 +54,12 @@
...
 
 # %%
-test_plots.Test_plots.test_plot_histograms_and_lagged_scatterplot1()
-
-# %%
+# Set input
+rng = np.random.default_rng(seed=0)
+samples = rng.normal(size=100)
+index = pd.date_range(start="2023-01-01", periods=len(samples), freq="D")
+srs = pd.Series(samples, index=index)
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/gallery_notebook.py:61]: so replace this one with `get_plot_histograms_and_lagged_scatterplot1()` from `core/plotting/test/test_plots.py`

File: core/plotting/notebooks/gallery_notebook.py (hunk 6 above)
Comment (DanilYachmenev) [reply core/plotting/notebooks/gallery_notebook.py]: see https://github.com/sorrentum/sorrentum/pull/402/commits/ddc77ce7a5e9808cba199a7a062f92a75787dfd8#r1260209236

but this is probably will be changed so good for now

File: core/plotting/notebooks/test/test_gallery_notebook.py (hunk 9 above)
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:35]: still trailing empty lines
remove them

File: core/plotting/test/test_plots.py [hunk 11]
```diff
@@ -8,20 +8,27 @@
...
+    """"
+    Helper method to be reused by gallery notebook. 
+    Returns sample series for `plot_histograms_and_lagged_scatterplot()`.
+    """
+    # Set inputs.
+    rng = np.random.default_rng(seed=0)
+    samples = rng.normal(size=100)
+    index = pd.date_range(start="2023-01-01", periods=len(samples), freq="D")
+    srs = pd.Series(samples, index=index)
+    return srs
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:22]: distance between independent functions and classes is usually 2 empty lines
did you run linter on this file? it should do it automatically

File: core/plotting/test/test_plots.py [hunk 12]
```diff
@@ -8,20 +8,27 @@
 
 _LOG = logging.getLogger(__name__)
 
+def get_plot_histograms_and_lagged_scatterplot1() -> pd.Series:
+    """"
+    Helper method to be reused by gallery notebook. 
+    Returns sample series for `plot_histograms_and_lagged_scatterplot()`.
+    """
+    # Set inputs.
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:16]: remove, not really clarifying anything here

File: core/plotting/test/test_plots.py [hunk 13]
```diff
@@ -8,20 +8,27 @@
 
 _LOG = logging.getLogger(__name__)
 
+def get_plot_histograms_and_lagged_scatterplot1() -> pd.Series:
+    """"
+    Helper method to be reused by gallery notebook. 
+    Returns sample series for `plot_histograms_and_lagged_scatterplot()`.
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:14]:
```
"""
Get a random Gaussian data series for plotting histograms and lagged scatterplot.
"""
```

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+    """
+    Empty config builder used by the test.
+    """
+    # We want to execute the notebook as it is, but config is needed
+    # so we pass an empty one.
+    config = {}
+    config = cconfig.Config()
+    config_list = cconfig.ConfigList([config])
+    return config_list
+
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:20]: here the distance should be maintained as well

File: core/plotting/notebooks/gallery_notebook.py [hunk 15]
```diff
@@ -57,12 +55,9 @@
 # ## `plot_histograms_and_lagged_scatterplot()`
 
 # %%
-# Set inputs.
-seq = np.concatenate([np.random.uniform(-1, 1, 100), np.random.choice([5, 10], 100)])
-index = pd.date_range(start="2023-01-01", periods=len(seq), freq="D")
-srs = pd.Series(seq, index=index)
+# Set input
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/gallery_notebook.py:58]: finish all the comments with dots

File: core/plotting/notebooks/test/test_gallery_notebook.py [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+@pytest.mark.superslow("~40 sec.")
+class TestGalleryNotebook(dsnrn.Test_Run_Notebook_TestCase):
+    def test_run_notebook(self) -> None:
+        """
+        Test if gallery notebook runs end-to-end with no errors.
+        """
+        notebook_path = "core/plotting/notebooks/gallery_notebook.ipynb"
+        config_func_path = "core.plotting.notebooks.test.test_gallery_notebook.build_config_list()"
+        # Run the notebook.
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:29]: remove this one, also not very clarifying comment

File: core/plotting/notebooks/gallery_notebook.py [hunk 17]
```diff
@@ -32,15 +32,13 @@
 import pandas as pd
 
 import core.plotting.visual_stationarity_test as cpvistte
+import core.plotting.test.test_plots as test_plots
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/gallery_notebook.py:35]: -> `... as cptetepl`

File: core/plotting/test/test_plots.py [hunk 18]
```diff
@@ -8,20 +8,27 @@
 
 _LOG = logging.getLogger(__name__)
 
+def get_plot_histograms_and_lagged_scatterplot1() -> pd.Series:
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:11]: following https://github.com/sorrentum/sorrentum/pull/393#discussion_r1261253727, make it as a static method of `Test_plots` class so basically as it was suggested in https://github.com/sorrentum/sorrentum/pull/393#discussion_r1258625891

File: core/plotting/notebooks/gallery_notebook.py (hunk 15 above)
Comment (DanilYachmenev) [reply core/plotting/notebooks/gallery_notebook.py:58]: not related to the headers ofc

**Comment by dchoi127** [comment]
My apologies. I need to get into the habit of running lint before committing changes. Thanks for your patience.

File: core/plotting/notebooks/test/test_gallery_notebook.py (hunk 7 above)
Comment (PomazkinG) [reply core/plotting/notebooks/test/test_gallery_notebook.py:24]: as discussed you guys can put `amp_path = hgit.get_amp_abs_path()` in the tests and Nina will refactor after the fact

File: core/plotting/notebooks/test/test_gallery_notebook.py
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/test_gallery_notebook.py:1]: we've changed an approach to dir structure for test notebook files (https://github.com/cryptokaizen/cmamp/issues/4775)

So move this file from  `core/plotting/notebooks/test` to `core/plotting/test`

File: core/plotting/notebooks/test/outcomes/result_0/run_notebook.0.log
Comment (DanilYachmenev) [inline core/plotting/notebooks/test/outcomes/result_0/run_notebook.0.log:1]: no need to commit log file, especially empty one

**Review by DanilYachmenev — COMMENTED** [review]
couple moments related to the recent changes to the unit testing notebooks approach

File: core/plotting/test/test_gallery_notebook.py [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+
+@pytest.mark.superslow("~40 sec.")
+class TestGalleryNotebook(dsnrn.Test_Run_Notebook_TestCase):
+    def test_run_notebook(self) -> None:
+        """
+        Test if gallery notebook runs end-to-end with no errors.
+        """
+        amp_path = hgit.get_amp_abs_path()
+        notebook_path = os.path.join(
+        amp_path, "core/plotting/notebooks/gallery_notebook.ipynb"
```
Comment (DanilYachmenev) [inline core/plotting/test/test_gallery_notebook.py:34]: run linter, this line should be intended

**Review by DanilYachmenev — COMMENTED** [review]
1 nit but apart from it lgtm
pls implement and then Ill move it to integrators

File: core/plotting/test/test_gallery_notebook.py (hunk 19 above)
Comment (DanilYachmenev) [reply core/plotting/test/test_gallery_notebook.py]: Develop a habit to run linter before each review request
it helps to handle the things you may overlook
https://github.com/sorrentum/sorrentum/pull/402/commits/7c3e4dbf88167ae5c1b3216b9dab33321a1f56a0

Comment (dchoi127) [reply core/plotting/test/test_gallery_notebook.py]: Understood. That is something I will work on. Thanks for your patience again.

**Comment by gpsaggese** [comment]
@dchoi127 can you pls resolve the conflicts so we can merge?

**Comment by dchoi127** [comment]
> @dchoi127 can you pls resolve the conflicts so we can merge?

My apologies didn't realize there were conflicts. Will learn for next time to be more aware of conflicts.  Resolving conflicts took a bit a time, trying to figure out nbdime and I realized that I can manually resolve changes. Hopefully didn't break anything. 

I appreciate your patience.
